lib/matplotlib/animation.py

```diff
@@ -946,8 +946,13 @@ def _start(self, *args):
         # First disconnect our draw event handler
         self._fig.canvas.mpl_disconnect(self._first_draw_id)
 
-        # Now do any initial draw
-        self._init_draw()
+        try:
+            # Now do any initial draw
+            self._init_draw()
+        except StopIteration:
```

Comment:
I think this might be ok, provided we use capturing this exception to either warn or raise another exception.

Reply:
Only took a year to get to!